src/include/stir/numerics/erfMapping.inl

```diff
@@ -0,0 +1,106 @@
+
+
+#include "stir/numerics/BSplines1DRegularGrid.h"
+
+START_NAMESPACE_STIR
+
+inline void
+erfMapping::
+setup()
+{
+  this->_sampling_period = (2 * this->maximum_sample_value) / this->get_num_samples();
+  
+  // Add two samples for end cases and compute a vector of erf values
+  std::vector<double> erf_values(this->get_num_samples() + 2);
+  for (int i=0; i<this->get_num_samples() + 2; ++i)
+    erf_values[i] = erf(i * this->_sampling_period - this->maximum_sample_value);
+
+  // Setup BSplines
+  BSpline::BSplines1DRegularGrid<double, double> spline(erf_values, BSpline::linear);
+  this->spline = spline;
+
+  erf_values_vec = erf_values;
+//  this->_is_setup = true;
+}
+
+inline double
+erfMapping::get_erf_BSplines_interpolation(double xp) const
+{
+#if 1
+    xp = std::clamp(xp,-this->maximum_sample_value, this->maximum_sample_value);
+#else
+  if (xp > this->maximum_sample_value)
+    return 1.0;
+
+  else if (xp < -this->maximum_sample_value)
+    return -1.0;
+#endif
+  return this->spline.BSplines((xp + this->maximum_sample_value) / this->_sampling_period);
+}
+
+
+inline double
+erfMapping::
+get_erf_linear_interpolation(double xp) const
+{
+#if 1
+    xp = std::clamp(xp,-this->maximum_sample_value, this->maximum_sample_value);
+#else
+  if (xp > this->maximum_sample_value)
+    return 1.0;
+
+  else if (xp < -this->maximum_sample_value)
+    return -1.0;
+#endif
+  // Find xp in index sequence
+  double xp_in_index = ((xp + this->maximum_sample_value) / this->_sampling_period);
+
+  // Find lower integer in index space
+  int lower = static_cast<int>(floor(xp_in_index));
+
+  // Linear interpolation of xp between vec[lower] and vec[lower + 1]
+  return erf_values_vec[lower] + (xp_in_index - lower) * (erf_values_vec[lower + 1] - erf_values_vec[lower]);
+}
+
+inline double
+erfMapping::get_erf_nearest_neighbour_interpolation(double xp) const
+{
+#if 1
+  xp = std::clamp(xp,-this->maximum_sample_value, this->maximum_sample_value);
+#else
+  if (xp > this->maximum_sample_value)
+    return 1.0;
+
+  else if (xp < -this->maximum_sample_value)
+    return -1.0;
+#endif
+  // Selects index of the nearest neighbour via rounding
+    return erf_values_vec[static_cast<int>((xp + this->maximum_sample_value) / this->_sampling_period)];
```

Comment:
static_cast doesn't round.. you'd have to use `stir::round`. will also have to think about out-of-range